2007-06-24  Loic Minier  <lool@dooz.org>
    
    	* gtk/Makefile.am (gtk_private_h_sources, gtk_semi_private_h_sources):
    	export gtkquery.h and gtksearchengine.h as semi-private
    	* gtk/gtkquery.h: add GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED #ifdef guard
    	* gtk/gtksearchengine.h: add GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED #ifdef
    	guard

2007-02-07  Kristian Rietveld  <kris@imendio.com>
    
    	* gtk/gtkfilechooserutils.[ch]
    	(hildon_gtk_file_chooser_install_properties): conditionalize with
    	MAEMO_CHANGES, port over doc comment.

2006-12-04  Tommi Komulainen  <tommi.komulainen@nokia.com>
    
    	Export enough symbols for implementing GtkFileChooser interface
    	outside gtk.
    
    	* gtk/gtk.symbols: add hildon_gtk_file_chooser_install_properties
    	* gtk/Makefile.am (INCLUDES): add -DGTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
    	(gtk_private_h_sources, gtk_semi_private_h_sources): export
    	gtkfilesystemmodel.h, gtkfilechooserprivate.h, and
    	gtkfilechooserutils.h as semi-private
    	* gtk/gtkfilechooserutils.c
    	(hildon_gtk_file_chooser_install_properties): add exported version of
    	_gtk_file_chooser_install_properties
    	* gtk/gtkfilesystemmodel.h: add GTK_FILE_SYSTEM_ENABLE_UNSUPPORTED
    	#ifdef guard
    	* gtk/gtkfilechooserprivate.h: add GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
    	#ifdef guard
    	* gtk/gtkfilechooserutils.h: add GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
    	#ifdef guard, declare hildon_gtk_file_chooser_install_properties
    
    	* tests/autotestfilechooser.c: build with
    	GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
Index: gtk+2.0-2.21.6/gtk/Makefile.am
===================================================================
--- gtk+2.0-2.21.6.orig/gtk/Makefile.am	2010-08-16 22:57:55.000000000 +0200
+++ gtk+2.0-2.21.6/gtk/Makefile.am	2010-08-17 11:51:57.600996862 +0200
@@ -49,6 +49,8 @@
 	-I$(top_srcdir) -I../gdk			\
 	-I$(top_srcdir)/gdk				\
 	-DGDK_DISABLE_DEPRECATED			\
+	-DMAEMO_CHANGES					\
+	-DGTK_FILE_CHOOSER_ENABLE_UNSUPPORTED		\
 	-DGTK_DISABLE_DEPRECATED			\
 	-DGTK_FILE_SYSTEM_ENABLE_UNSUPPORTED		\
 	-DGTK_PRINT_BACKEND_ENABLE_UNSUPPORTED		\
@@ -356,12 +358,16 @@
 # Installed header files without compatibility guarantees
 # that are not included in gtk/gtk.h
 gtk_semi_private_h_sources =    \
-	gtktextlayout.h
+	gtktextlayout.h         \
+	gtkfilesystem.h		\
+	gtkfilesystemmodel.h	\
+	gtkfilechooserprivate.h	\
+	gtkfilechooserutils.h	\
+	gtkquery.h		\
+	gtksearchengine.h
 
 # GTK+ header files that don't get installed
 gtk_private_h_sources =		\
-	gtkquery.h		\
-	gtksearchengine.h	\
 	gtksearchenginesimple.h	\
 	gtkdndcursors.h		\
 	gtkentryprivate.h	\
@@ -370,11 +376,7 @@
 	gtkfilechooserdefault.h	\
 	gtkfilechooserembed.h	\
 	gtkfilechooserentry.h	\
-	gtkfilechooserprivate.h	\
 	gtkfilechoosersettings.h \
-	gtkfilechooserutils.h	\
-	gtkfilesystem.h		\
-	gtkfilesystemmodel.h	\
 	gtkiconcache.h		\
 	gtkintl.h		\
 	gtkkeyhash.h		\
Index: gtk+2.0-2.21.6/gtk/gtk.symbols
===================================================================
--- gtk+2.0-2.21.6.orig/gtk/gtk.symbols	2010-08-16 22:57:55.000000000 +0200
+++ gtk+2.0-2.21.6/gtk/gtk.symbols	2010-08-17 11:51:57.600996862 +0200
@@ -1641,6 +1641,12 @@
 #endif
 #endif
 
+#if IN_HEADER(__GTK_FILE_CHOOSER_UTILS_H__)
+#if IN_FILE(__GTK_FILE_CHOOSER_UTILS_C__)
+hildon_gtk_file_chooser_install_properties
+#endif
+#endif
+
 #if IN_HEADER(__GTK_FILE_CHOOSER_WIDGET_H__)
 #if IN_FILE(__GTK_FILE_CHOOSER_WIDGET_C__)
 gtk_file_chooser_widget_get_type G_GNUC_CONST
Index: gtk+2.0-2.21.6/gtk/gtkfilechooserprivate.h
===================================================================
--- gtk+2.0-2.21.6.orig/gtk/gtkfilechooserprivate.h	2010-08-16 22:57:55.000000000 +0200
+++ gtk+2.0-2.21.6/gtk/gtkfilechooserprivate.h	2010-08-17 11:51:57.819997600 +0200
@@ -21,6 +21,14 @@
 #ifndef __GTK_FILE_CHOOSER_PRIVATE_H__
 #define __GTK_FILE_CHOOSER_PRIVATE_H__
 
+/* This is a "semi-private" header; it is meant only for
+ * alternate GtkFileChooser implementations; no stability guarantees 
+ * are made at this point
+ */
+#ifndef GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
+#error "gtkfilechooserprivate.h is not supported API for general use"
+#endif
+
 #include "gtkfilechooser.h"
 #include "gtkfilesystem.h"
 #include "gtkfilesystemmodel.h"
Index: gtk+2.0-2.21.6/gtk/gtkfilechooserutils.c
===================================================================
--- gtk+2.0-2.21.6.orig/gtk/gtkfilechooserutils.c	2010-08-16 22:57:55.000000000 +0200
+++ gtk+2.0-2.21.6/gtk/gtkfilechooserutils.c	2010-08-17 11:51:57.819997600 +0200
@@ -122,6 +122,22 @@
 				    "create-folders");
 }
 
+#ifdef MAEMO_CHANGES
+/**
+ * hildon_gtk_file_chooser_install_properties:
+ *
+ * Exactly the same as the private _gtk_file_chooser_install_properties()
+ * but exported for hildon-fm.
+ *
+ * Since: maemo 2.0
+ */
+void
+hildon_gtk_file_chooser_install_properties (GObjectClass *klass)
+{
+  _gtk_file_chooser_install_properties (klass);
+}
+#endif /* MAEMO_CHANGES */
+
 /**
  * _gtk_file_chooser_delegate_iface_init:
  * @iface: a #GtkFileChoserIface structure
@@ -363,3 +379,6 @@
   g_signal_emit_by_name (data, "confirm-overwrite", &conf);
   return conf;
 }
+
+#define __GTK_FILE_CHOOSER_UTILS_C__
+#include "gtkaliasdef.c"
Index: gtk+2.0-2.21.6/gtk/gtkfilechooserutils.h
===================================================================
--- gtk+2.0-2.21.6.orig/gtk/gtkfilechooserutils.h	2010-08-16 22:57:55.000000000 +0200
+++ gtk+2.0-2.21.6/gtk/gtkfilechooserutils.h	2010-08-17 11:51:57.819997600 +0200
@@ -22,6 +22,14 @@
 #ifndef __GTK_FILE_CHOOSER_UTILS_H__
 #define __GTK_FILE_CHOOSER_UTILS_H__
 
+/* This is a "semi-private" header; it is meant only for
+ * alternate GtkFileChooser implementations; no stability guarantees 
+ * are made at this point
+ */
+#ifndef GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
+#error "gtkfilechooserutils.h is not supported API for general use"
+#endif
+
 #include "gtkfilechooserprivate.h"
 
 G_BEGIN_DECLS
@@ -46,6 +54,9 @@
 } GtkFileChooserProp;
 
 void _gtk_file_chooser_install_properties (GObjectClass *klass);
+#ifdef MAEMO_CHANGES
+void hildon_gtk_file_chooser_install_properties (GObjectClass *klass);
+#endif /* MAEMO_CHANGES */
 
 void _gtk_file_chooser_delegate_iface_init (GtkFileChooserIface *iface);
 void _gtk_file_chooser_set_delegate        (GtkFileChooser *receiver,
Index: gtk+2.0-2.21.6/gtk/gtkfilesystemmodel.h
===================================================================
--- gtk+2.0-2.21.6.orig/gtk/gtkfilesystemmodel.h	2010-06-16 17:27:02.000000000 +0200
+++ gtk+2.0-2.21.6/gtk/gtkfilesystemmodel.h	2010-08-17 11:51:57.823997341 +0200
@@ -21,6 +21,10 @@
 #ifndef __GTK_FILE_SYSTEM_MODEL_H__
 #define __GTK_FILE_SYSTEM_MODEL_H__
 
+#ifndef GTK_FILE_SYSTEM_ENABLE_UNSUPPORTED
+#error "GtkFileSystemModel is not supported API for general use"
+#endif
+
 #include <gio/gio.h>
 #include <gtk/gtkfilefilter.h>
 #include <gtk/gtktreemodel.h>
Index: gtk+2.0-2.21.6/gtk/gtkquery.h
===================================================================
--- gtk+2.0-2.21.6.orig/gtk/gtkquery.h	2009-04-02 15:54:56.000000000 +0200
+++ gtk+2.0-2.21.6/gtk/gtkquery.h	2010-08-17 11:51:57.823997341 +0200
@@ -23,6 +23,14 @@
 #ifndef __GTK_QUERY_H__
 #define __GTK_QUERY_H__
 
+/* This is a "semi-private" header; it is meant only for
+ * alternate GtkFileChooser implementations; no stability guarantees 
+ * are made at this point
+ */
+#ifndef GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
+#error "gtkfilechooserprivate.h is not supported API for general use"
+#endif
+
 #include <glib-object.h>
 
 G_BEGIN_DECLS
Index: gtk+2.0-2.21.6/gtk/gtksearchengine.h
===================================================================
--- gtk+2.0-2.21.6.orig/gtk/gtksearchengine.h	2010-04-09 04:29:58.000000000 +0200
+++ gtk+2.0-2.21.6/gtk/gtksearchengine.h	2010-08-17 11:51:57.873067895 +0200
@@ -23,6 +23,14 @@
 #ifndef __GTK_SEARCH_ENGINE_H__
 #define __GTK_SEARCH_ENGINE_H__
 
+/* This is a "semi-private" header; it is meant only for
+ * alternate GtkFileChooser implementations; no stability guarantees 
+ * are made at this point
+ */
+#ifndef GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
+#error "gtkfilechooserprivate.h is not supported API for general use"
+#endif
+
 #include "gtkquery.h"
 
 G_BEGIN_DECLS
